packages/client-engine-runtime/src/interpreter/QueryInterpreter.ts

```diff
@@ -79,11 +79,9 @@ export class QueryInterpreter {
 
       case 'let': {
         const nestedScope: ScopeBindings = Object.create(scope)
-        await Promise.all(
-          node.args.bindings.map(async (binding) => {
-            nestedScope[binding.name] = await this.interpretNode(binding.expr, queryable, scope, generators)
-          }),
-        )
+        for (const binding of node.args.bindings) {
+          nestedScope[binding.name] = await this.interpretNode(binding.expr, queryable, nestedScope, generators)
+        }
```

Comment:
Important change: bindings can now reference previous bindings in the same `let` expression:

```
let a = ...;
    b = f a
in ...
```